### PR TITLE
Added HoverProvider to IntelliSense features

### PR DESCRIPTION
**Pull request by tdusnoki**
IoT.js-VSCode-DCO-1.0-Signed-off-by: Tibor Dusnoki tdusnoki@inf.u-szeged.hu

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -163,6 +163,33 @@ const createModules = (document: vscode.TextDocument, position: vscode.Position)
...
+  let hoverText: string;
+  let hoverModule: string;
+  const availableModules = defaultModules.concat(lookForModules(document.getText()));
+
+  availableModules.forEach(mod => {
+    if ( match[1] === mod.link) {
+      hoverText = text;
+      hoverModule = mod.mod;
+    }
+  });
```
Comment (knightburton) [inline src/extension.ts]: I would use a similar pattern like this over here:
```typescript
const { hoverText, hoverModule } = availableModules.find(mod => mod.link === match[1]).map(mod => ({
  hoverText: text,
  hoverModule: mod.mod
});
```
in tis case we can eliminate the `let`.

File: src/extension.ts [hunk 2]
```diff
@@ -163,6 +163,33 @@ const createModules = (document: vscode.TextDocument, position: vscode.Position)
   return [];
 };
 
+const createHover = (document: vscode.TextDocument, position: vscode.Position): vscode.Hover => {
+  const text = document.getText(document.getWordRangeAtPosition(position));
+  const rm = new RegExp('([a-zA-Z0-9$_ ]+)\\.' + text + '([a-zA-Z0-9$_ ]*)');
+  const match = rm.exec(document.lineAt(position.line).text);
+  const modules = Object.keys(iotjs);
+  let hoverContent = Array<vscode.MarkdownString>();
```
Comment (knightburton) [inline src/extension.ts]: let hoverContent: vscode.MarkdownString[] = [];

**Review by knightburton — CHANGES_REQUESTED** [review]

**Review by knightburton — APPROVED** [review]
LGTM

**Review by yichoi — APPROVED** [review]
LGTM

**Pull request merged**